### PR TITLE
[KEYBOARD] Fix failing tests added in 6aacfa93 and c1c12793

### PR DESCRIPTION
**Pull request by binarymaster**
This also should fix related problems with "F17" shortcut key in the menu items of some programs.

Dedicated to @julcar. This PR effectively supersedes #4730.

JIRA issues: [CORE-17906](https://jira.reactos.org/browse/CORE-17906) and [CORE-3903](https://jira.reactos.org/browse/CORE-3903)

Note there are also misleading comments about the offsets in `scancode_to_vk` array, but I will probably fix that later (and not by hand).

## Test results

- **VirtualKey**: one failure less
- **KbdLayout**: all failures fixed
 
```sh
VirtualKey.c:52: Test failed: [4] ScanCode = 0, expected 29

VirtualKey: 18 tests executed (0 marked as todo, 1 failure), 0 skipped.

KbdLayout.c:23: Testing 'kbda1.dll'...
KbdLayout.c:23: Testing 'kbda2.dll'...
KbdLayout.c:23: Testing 'kbda3.dll'...
KbdLayout.c:23: Testing 'kbdal.dll'...
KbdLayout.c:23: Testing 'kbdarme.dll'...
KbdLayout.c:23: Testing 'kbdarmw.dll'...
KbdLayout.c:23: Testing 'kbdaze.dll'...
KbdLayout.c:23: Testing 'kbdazel.dll'...
KbdLayout.c:23: Testing 'kbdbe.dll'...
KbdLayout.c:23: Testing 'kbdbga.dll'...
KbdLayout.c:23: Testing 'kbdbgt.dll'...
KbdLayout.c:23: Testing 'kbdblr.dll'...
KbdLayout.c:23: Testing 'kbdbr.dll'...
KbdLayout.c:23: Testing 'kbdbu.dll'...
KbdLayout.c:23: Testing 'kbdbur.dll'...
KbdLayout.c:23: Testing 'kbdcan.dll'...
KbdLayout.c:23: Testing 'kbdcr.dll'...
KbdLayout.c:23: Testing 'kbdcz.dll'...
KbdLayout.c:23: Testing 'kbdcz1.dll'...
KbdLayout.c:23: Testing 'kbdda.dll'...
KbdLayout.c:23: Testing 'kbddv.dll'...
KbdLayout.c:23: Testing 'kbdeo.dll'...
KbdLayout.c:23: Testing 'kbdest.dll'...
KbdLayout.c:23: Testing 'kbdfc.dll'...
KbdLayout.c:23: Testing 'kbdfi.dll'...
KbdLayout.c:23: Testing 'kbdfr.dll'...
KbdLayout.c:23: Testing 'kbdgeo.dll'...
KbdLayout.c:23: Testing 'kbdgerg.dll'...
KbdLayout.c:23: Testing 'kbdgneo.dll'...
KbdLayout.c:23: Testing 'kbdgr.dll'...
KbdLayout.c:23: Testing 'kbdgr1.dll'...
KbdLayout.c:23: Testing 'kbdhe.dll'...
KbdLayout.c:23: Testing 'kbdheb.dll'...
KbdLayout.c:23: Testing 'kbdhu.dll'...
KbdLayout.c:23: Testing 'kbdic.dll'...
KbdLayout.c:23: Testing 'kbdinasa.dll'...
KbdLayout.c:23: Testing 'kbdinben.dll'...
KbdLayout.c:23: Testing 'kbdindev.dll'...
KbdLayout.c:23: Testing 'kbdinguj.dll'...
KbdLayout.c:23: Testing 'kbdinmal.dll'...
KbdLayout.c:23: Testing 'kbdir.dll'...
KbdLayout.c:23: Testing 'kbdit.dll'...
KbdLayout.c:23: Testing 'kbdjpn.dll'...
KbdLayout.c:23: Testing 'kbdkaz.dll'...
KbdLayout.c:23: Testing 'kbdkor.dll'...
KbdLayout.c:23: Testing 'kbdla.dll'...
KbdLayout.c:23: Testing 'kbdlt1.dll'...
KbdLayout.c:23: Testing 'kbdlv.dll'...
KbdLayout.c:23: Testing 'kbdmac.dll'...
KbdLayout.c:23: Testing 'kbdne.dll'...
KbdLayout.c:23: Testing 'kbdno.dll'...
KbdLayout.c:23: Testing 'kbdpl.dll'...
KbdLayout.c:23: Testing 'kbdpl1.dll'...
KbdLayout.c:23: Testing 'kbdpo.dll'...
KbdLayout.c:23: Testing 'kbdro.dll'...
KbdLayout.c:23: Testing 'kbdrost.dll'...
KbdLayout.c:23: Testing 'kbdru.dll'...
KbdLayout.c:23: Testing 'kbdru1.dll'...
KbdLayout.c:23: Testing 'kbdsf.dll'...
KbdLayout.c:23: Testing 'kbdsg.dll'...
KbdLayout.c:23: Testing 'kbdsl.dll'...
KbdLayout.c:23: Testing 'kbdsl1.dll'...
KbdLayout.c:23: Testing 'kbdsp.dll'...
KbdLayout.c:23: Testing 'kbdsw.dll'...
KbdLayout.c:23: Testing 'kbdtat.dll'...
KbdLayout.c:23: Testing 'kbdth0.dll'...
KbdLayout.c:23: Testing 'kbdth1.dll'...
KbdLayout.c:23: Testing 'kbdth2.dll'...
KbdLayout.c:23: Testing 'kbdth3.dll'...
KbdLayout.c:23: Testing 'kbdtuf.dll'...
KbdLayout.c:23: Testing 'kbdtuq.dll'...
KbdLayout.c:23: Testing 'kbduk.dll'...
KbdLayout.c:23: Testing 'kbdur.dll'...
KbdLayout.c:23: Testing 'kbdurs.dll'...
KbdLayout.c:23: Testing 'kbdus.dll'...
KbdLayout.c:23: Testing 'kbdusa.dll'...
KbdLayout.c:23: Testing 'kbdusl.dll'...
KbdLayout.c:23: Testing 'kbdusr.dll'...
KbdLayout.c:23: Testing 'kbdusx.dll'...
KbdLayout.c:23: Testing 'kbduzb.dll'...
KbdLayout.c:23: Testing 'kbdvntc.dll'...
KbdLayout.c:23: Testing 'kbdycc.dll'...
KbdLayout.c:23: Testing 'kbdycl.dll'...

KbdLayout: 83 tests executed (0 marked as todo, 0 failures), 0 skipped.
```

### CODE REVIEW

**Review by JoachimHenze — APPROVED** [review]

File: dll/keyboard/kbdbe/kbdbe.c [hunk 1]
```diff
@@ -98,9 +98,6 @@ ROSDATA USHORT scancode_to_vk[] = {
   /* - 77 - */
   VK_EMPTY, VK_EMPTY, VK_EMPTY, VK_EMPTY,
   VK_EMPTY, VK_EMPTY, VK_EMPTY, VK_EMPTY, /* PA1 */
-  VK_EMPTY,
```
Comment (HBelusca) [inline dll/keyboard/kbdbe/kbdbe.c:101]: It's different from the previous layouts?

File: dll/keyboard/kbdcz/kbdcz.c [hunk 2]
```diff
@@ -98,9 +98,6 @@ ROSDATA USHORT scancode_to_vk[] = {
   /* - 77 - */
   VK_EMPTY, VK_EMPTY, VK_EMPTY, VK_EMPTY,
   VK_EMPTY, VK_EMPTY, VK_EMPTY, VK_EMPTY, /* PA1 */
-  VK_EMPTY,
```
Comment (HBelusca) [inline dll/keyboard/kbdcz/kbdcz.c:101]: Ditto

File: dll/keyboard/kbdit/kbdit.c [hunk 3]
```diff
@@ -155,9 +155,6 @@ ROSDATA USHORT scancode_to_vk[] = {
 /* 7c */  VK_EMPTY,
 /* 7d */  VK_EMPTY,
 /* 7e */  VK_EMPTY,
-/* 7f */  VK_EMPTY,
```
Comment (HBelusca) [inline dll/keyboard/kbdit/kbdit.c:158]: Interesting that it stops at 7e, and 7f is not even "empty" xd

**Review by HBelusca — APPROVED** [review]

File: dll/keyboard/kbdbe/kbdbe.c (hunk 1 above)
Comment (binarymaster) [reply dll/keyboard/kbdbe/kbdbe.c:101]: Yes, there are some differences in different layouts. Some had multiple zeros in array, some only in the end.

File: dll/keyboard/kbdcz/kbdcz.c (hunk 2 above)
Comment (binarymaster) [reply dll/keyboard/kbdcz/kbdcz.c:101]: Same as above, I just removed excessive values (there should be 127 maximum), and replaced zeros with VK_EMPTY.

File: dll/keyboard/kbdit/kbdit.c (hunk 3 above)
Comment (binarymaster) [reply dll/keyboard/kbdit/kbdit.c:158]: Yeah it's just array of values, so zero-termination is not needed here :slightly_smiling_face:

**Comment by JoachimHenze** [comment]
Since the PR number was not mentioned in the commit msg to master, I forgot to mention it in the backport as well. Therefore added here again:

master fixed by [0.4.15-dev-5145-gcdf9070|https://github.com/reactos/reactos/commit/cdf907074d607a2025d4e099bc823286e24c8be4] (2022-10-05)
and ported back together with other related fixes into:
[0.4.14-release-40-g1be3db2|https://github.com/reactos/reactos/commit/1be3db2baa1d1ca444f4e43f1565d41fdcc62bb4] (2022-12-21)
[0.4.13-release-113-ge773a38|https://github.com/reactos/reactos/commit/e773a38d7b30b76ec27abe3e82bf6ed16ad164f9] (2022-12-21)
[0.4.12-release-143-g37be0cf|https://github.com/reactos/reactos/commit/37be0cfc067ff04a81b699c24d2df98def918c13] (2022-12-21)
[0.4.11-release-153-g330d3bd|https://github.com/reactos/reactos/commit/330d3bd54c29d03e3b07998411946d0b15076a9b] (2022-12-21)
[0.4.10-release-169-g300a7b5|https://github.com/reactos/reactos/commit/300a7b52108feb1952bce65c206f600e2af21f1a] (2022-12-21)
[0.4.9-release-180-g4f64a0c|https://github.com/reactos/reactos/commit/4f64a0cde371b14a8caeb6b521661a384f776a85] (2022-12-21)
[0.4.8-release-186-gaa8c1a4|https://github.com/reactos/reactos/commit/aa8c1a424d8f845fa9a98e78516a9179b8cd692c] (2022-12-21)
[0.4.7-release-209-g36741a8|https://github.com/reactos/reactos/commit/36741a8569c65ec68471ab0f24b885086cec9b35] (2022-12-21)

**Pull request merged**